src/poetry/core/packages/package.py

```diff
@@ -559,7 +559,11 @@ def clone(self: T) -> T:
         return clone
 
     def __hash__(self) -> int:
-        return super().__hash__() ^ hash(self._version)
+        return (
+            super().__hash__()
+            ^ hash(self._version)
+            ^ hash(frozenset(self.extras.keys()))
```

Comment:
If we use `extras` in `__hash__`, we have to use it in `__eq__`, too. Actually, `extras` should probably not be used in hash because they can change and thus the hash...